### PR TITLE
Update pod ip using active container only.

### LINKED ISSUE #5997: HEAD: monitoring-heapster-controller container exit with error every 10 mins
While fixing the issue #5957 on describe events for pod, I found kubelet keep restarting monitoring-heapster-controller container:

```
$ cluster/kubectl.sh describe pods monitoring-heapster-controller-h3x9k
current-context: "golden-system-455_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl describe pods monitoring-heapster-controller-h3x9k
Name:				monitoring-heapster-controller-h3x9k
Image(s):			kubernetes/heapster:v0.9
Host:				kubernetes-minion-agug.c.golden-system-455.internal/104.197.6.91
Labels:				kubernetes.io/cluster-service=true,name=heapster,uses=monitoring-influxdb
Status:				Running
Replication Controllers:	monitoring-heapster-controller (1/1 replicas created)
Conditions:
  Type		Status
  Ready 	True 
Events:
  FirstSeen				LastSeen			Count	From								SubobjectPath			Reason	Message
  Thu, 26 Mar 2015 09:12:52 -0700	Thu, 26 Mar 2015 09:12:52 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	created	Created with docker id 8b29f11cb1f7187910668551cdd4037aa40af3b69536ed934c685d0704cf05dd
  Thu, 26 Mar 2015 09:12:52 -0700	Thu, 26 Mar 2015 09:12:52 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	started	Started with docker id 8b29f11cb1f7187910668551cdd4037aa40af3b69536ed934c685d0704cf05dd
  Thu, 26 Mar 2015 09:23:03 -0700	Thu, 26 Mar 2015 09:23:03 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	created	Created with docker id 27e1c4ee3ffc54e86b381af4f848f374308466fa606ff358b70ff687e07bb9d0
  Thu, 26 Mar 2015 09:23:03 -0700	Thu, 26 Mar 2015 09:23:03 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	started	Started with docker id 27e1c4ee3ffc54e86b381af4f848f374308466fa606ff358b70ff687e07bb9d0
  Thu, 26 Mar 2015 09:33:13 -0700	Thu, 26 Mar 2015 09:33:13 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	created	Created with docker id 29504f60785eccaafc208e56d43540bf738ed124dbc96ed3e57de4eab5318040
  Thu, 26 Mar 2015 09:33:13 -0700	Thu, 26 Mar 2015 09:33:13 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	started	Started with docker id 29504f60785eccaafc208e56d43540bf738ed124dbc96ed3e57de4eab5318040
  Thu, 26 Mar 2015 09:43:23 -0700	Thu, 26 Mar 2015 09:43:23 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	created	Created with docker id bd8a131a7382f4b0bcb125f3ab38aa933d7c7a438e9b4b428bdfbfcd3b69eaea
  Thu, 26 Mar 2015 09:43:23 -0700	Thu, 26 Mar 2015 09:43:23 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	started	Started with docker id bd8a131a7382f4b0bcb125f3ab38aa933d7c7a438e9b4b428bdfbfcd3b69eaea
  Thu, 26 Mar 2015 09:53:33 -0700	Thu, 26 Mar 2015 09:53:33 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	created	Created with docker id f8c1e1bd3024c2ceae2c32b7bfb0244e62a33945834d6299982e3a66678a1778
  Thu, 26 Mar 2015 09:53:34 -0700	Thu, 26 Mar 2015 09:53:34 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	started	Started with docker id f8c1e1bd3024c2ceae2c32b7bfb0244e62a33945834d6299982e3a66678a1778
  Thu, 26 Mar 2015 10:03:44 -0700	Thu, 26 Mar 2015 10:03:44 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	created	Created with docker id c5f1b1a2ce3c161ca380a5f4886ad93419947873f493eca4dde0f9724793708d
  Thu, 26 Mar 2015 10:03:44 -0700	Thu, 26 Mar 2015 10:03:44 -0700	1	{kubelet kubernetes-minion-agug.c.golden-system-455.internal}	spec.containers{heapster}	started	Started with docker id c5f1b1a2ce3c161ca380a5f4886ad93419947873f493eca4dde0f9724793708d

```

I retrieved the container log, and found it exit due to database creation failure:

```
I0326 17:03:44.287397       7 driver.go:164] Using influxdb on host "10.0.232.139:80" with database "k8s"
E0326 17:03:44.490457       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: net/http: transport closed before response was received. Retrying after 30 seconds
E0326 17:04:14.491752       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:04:44.493130       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:05:14.494835       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:05:44.496148       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:06:14.497545       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:06:44.499208       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:07:14.502519       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:07:44.504063       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:08:14.505358       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:08:44.730301       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: net/http: transport closed before response was received. Retrying after 30 seconds
E0326 17:09:14.732977       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:09:44.836341       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: net/http: transport closed before response was received. Retrying after 30 seconds
E0326 17:10:14.937756       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:10:45.142047       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: net/http: transport closed before response was received. Retrying after 30 seconds
E0326 17:11:15.143812       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:11:45.146168       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:12:15.147444       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:12:45.148914       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: read tcp 10.0.232.139:80: connection reset by peer. Retrying after 30 seconds
E0326 17:13:15.254122       7 driver.go:175] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: net/http: transport closed before response was received. Retrying after 30 seconds
E0326 17:13:44.442688       7 reflector.go:118] watch of *api.Pod ended with error: 401: The event in requested index is outdated and cleared (the requested history has been cleared [237511/237343]) [238510]
E0326 17:13:45.254457       7 heapster.go:52] Database creation failed: Post http://10.0.232.139:80/db?u=root&p=root: net/http: transport closed before response was received
```

I didn't follow up on why influxdb issue. cc/ @vishh

### PR DESCRIPTION
**Pull request by vishh**
Fixes #5997

### CODE REVIEW

File: pkg/kubelet/dockertools/docker.go [hunk 1]
```diff
@@ -685,6 +687,9 @@ func GetDockerPodStatus(client DockerInterface, manifest api.PodSpec, podFullNam
 		}
 	}
 
+	if podStatus.PodIP == "" {
+		return nil, fmt.Errorf("pod infra container not found. Pod IP is not populated.")
```
Comment (dchen1107) [inline pkg/kubelet/dockertools/docker.go]: using ErrNoContainersInPod instead?

**Comment by dchen1107** [comment]
LGTM except one small nit.

There is definitely issue in the logic of  Post PodStatus on change, we need keep an eye on that.

File: pkg/kubelet/dockertools/docker.go (hunk 1 above)
Comment (vishh) [inline pkg/kubelet/dockertools/docker.go]: Done

**Comment by vishh** [comment]
Self merging as per @dchen1107's suggestion.

**Pull request merged**